control-plane/catalog/to-consul/resource.go

```diff
@@ -511,6 +511,18 @@ func (t *ServiceResource) generateRegistrations(key string) {
 			r.Service = &rs
 			r.Service.ID = serviceID(r.Service.Service, ip)
 			r.Service.Address = ip
+			// Adding information about service weight.
+			// Overrides the existing weight if present
+			if weight, ok := svc.Annotations[annotationServiceWeight]; ok || weight != "" {
+				err := t.setServiceWeight(weight, len(ips), r)
```

Comment:
Should this be `&r`?

Reply:
What is the rational behind using `&r`? I came from java background so help me understand.

Reply:
Its about pass by reference. If tests are passing let it be.

Reply:
I guess it should be `&r` if you are mutating attributes of `r`. In the current state, the struct gets passed by value semantics

See https://go.dev/play/p/qX0awpC_yY9 and https://go.dev/play/p/OmmrCWLBhGh